### PR TITLE
add :keys opts with default false to stringify

### PR DESCRIPTION
**Pull request by sitch**

### CODE REVIEW

**Comment by sitch** [comment]
https://github.com/am-kantox/elixir-iteraptor/issues/11

File: lib/iteraptor.ex [hunk 1]
```diff
@@ -435,15 +438,18 @@ defmodule Iteraptor do
       input,
       fn
         {k, [{_, _} | _] = kw} when is_list(k) ->
-          {k |> List.last() |> to_string(), jsonify(kw, opts)}
+          {k |> List.last() |> do_stringify(opts), jsonify(kw, opts)}
```
Comment (am-kantox) [inline lib/iteraptor.ex]: I would avoid repetitive `Keyword.get/3` calls and would do instead:

```elixir
  stringify_keys = Keyword.get(opts, :keys, true)
  Iteraptor.map(
    ...
    {k |> List.last() |> do_stringify(stringify_keys), jsonify(kw, opts)}
```

and see below.

File: lib/iteraptor.ex [hunk 2]
```diff
@@ -435,15 +438,18 @@ defmodule Iteraptor do
...
-          {k |> List.last() |> to_string(), jsonify(v, opts)}
+          {k |> List.last() |> do_stringify(opts), jsonify(v, opts)}
       end,
       yield: :all
     )
   end
 
+  defp do_stringify(k, opts),
+    do: if(Keyword.get(opts, :keys, true), do: to_string(k), else: k)
+
```
Comment (am-kantox) [inline lib/iteraptor.ex]:
```suggestion
  @spec do_stringify(any(), boolean()) :: any() | binary()
  defp do_stringify(k, false), do: k
  defp do_stringify(k, _) when is_atom(k), do: Atom.to_string(k) # faster
  defp do_stringify(k, _), do: to_string(k)
```

**Review by am-kantox — CHANGES_REQUESTED** [review]

**Comment by sitch** [comment]
Merged them!

**Comment by sitch** [comment]
One other thing would be the typespec should technically change to `t:map/0` wasn't sure what you thought

**Review by am-kantox — APPROVED** [review]
:zap: :droplet: :zap: :gem: :zap: :blue_heart: :zap:

**Pull request merged**